geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/MergedDataStoreView.scala

```diff
@@ -95,13 +96,15 @@ class MergedDataStoreView(val stores: Seq[DataStore], namespace: Option[String]
 
   override def getFeatureSource(name: Name): SimpleFeatureSource = getFeatureSource(name.getLocalPart)
 
-  override def getFeatureSource(typeName: String): SimpleFeatureSource =
-    new MergedFeatureSourceView(this, stores.map(_.getFeatureSource(typeName)), getSchema(typeName))
+  override def getFeatureSource(typeName: String): SimpleFeatureSource = {
+    val sources = stores.map { case (store, filter) => (store.getFeatureSource(typeName), filter) }
```

Comment:
What happens if the desired featureType is only available in some datastores?

Reply:
in getSchema and getTypeNames, it should only return the feature types common in each store. So if you request one that doesn't exist in one of the stores, you'll usually get a schema doesn't exist exception from that store.